### PR TITLE
Chores/changes in container

### PR DESCRIPTION
**Pull request by kurbaitis**

### CODE REVIEW

File: lib/hammer/container.rb [hunk 1]
```diff
@@ -3,7 +3,14 @@ class Container < Struct.new(:controller)
...
       super
-      self.class.send(:attr_writer, *attributes_for_read)
+      self.class.send(:attr_writer, *methods_to_bind)
+    end
+
+    def methods_to_bind
+      ancestors = self.class.ancestors.select { |ancestor| ancestor.name.try(:match, /Container$/) }
+      methods = ancestors.map { |ancestor| ancestor.public_instance_methods(false) }.flatten
+
+      methods.select { |method| !method.to_s.match(/=$/) }
```
Comment (alaibe) [inline lib/hammer/container.rb]: Could you explain:
* Why not private
* Why this change?

File: lib/hammer/container.rb [hunk 2]
```diff
@@ -3,7 +3,14 @@ class Container < Struct.new(:controller)
 
     def initialize(*args)
       super
-      self.class.send(:attr_writer, *attributes_for_read)
+      self.class.send(:attr_writer, *methods_to_bind)
+    end
+
+    def methods_to_bind
+      ancestors = self.class.ancestors.select { |ancestor| ancestor.name.try(:match, /Container$/) }
+      methods = ancestors.map { |ancestor| ancestor.public_instance_methods(false) }.flatten
```
Comment (alaibe) [inline lib/hammer/container.rb]: flat_map

File: lib/hammer/container.rb (hunk 1 above)
Comment (kurbaitis) [inline lib/hammer/container.rb]: 1. used by concern to delegate
2. to allow methods from ancestors that class names ends with Container
